frappe/website/path_resolver.py

```diff
@@ -124,14 +126,14 @@ def resolve_redirect(path, query_string=None):
 
 		try:
 			match = re.match(pattern, path_to_match)
-		except re.error as e:
+		except re.error:
 			frappe.log_error("Broken Redirect: " + pattern)
 
 		if match:
 			redirect_to = re.sub(pattern, rule["target"], path_to_match)
 			frappe.flags.redirect_location = redirect_to
 			frappe.cache.hset("website_redirects", path_to_match, redirect_to)
-			raise frappe.Redirect
+			raise frappe.Redirect(rule.get("redirect_http_status", 301))
```

Comment:
Suggestion: Also cache the status code

```python
		if match:
			redirect_to = re.sub(pattern, rule["target"], path_to_match)
			redirect_http_status = rule.get("redirect_http_status") or 301
			frappe.flags.redirect_location = redirect_to
			frappe.cache.hset("website_redirects", path_to_match, (redirect_to, redirect_http_status))
			raise frappe.Redirect(redirect_http_status)
```

Reply:
Makes sense, thanks for pointing out
2224d86c9487587606d5894a642fd7b06511fd9e